R/estimate_contrasts.R

```diff
@@ -11,9 +11,32 @@
 #'   "bonferroni", "BH", "BY", "fdr" or "none". See the p-value adjustment
 #'   section in the `emmeans::test` documentation.
 #' @param adjust Deprecated in favour of `p_adjust`.
+#' @param effectsize Desired measure of standardized effect size, one of "none"
+#' (default), "emmeans", "bootES".
+#' @param bootES_type Specifies the type of effect-size measure to
+#' estimate when using `effectsize = "bootES"`. One of `c("unstandardized",
+#' "cohens.d", "hedges.g", "cohens.d.sigma", "r", "akp.robust.d")`. See`
+#' effect.type` argument of [bootES::bootES] for details.
+#' @param bootstraps The number of bootstrap resamples to perform.
 #'
 #' @inherit estimate_slopes details
 #'
+#' @section Effect Size: By default, `estimate_contrasts` reports no standardized effect size
+#' on purpose. Should one request one, some things are to keep in mind. As the
+#' authors of `emmeans` write, "There is substantial disagreement among
+#' practitioners on what is the appropriate sigma to use in computing effect
+#' sizes; or, indeed, whether any effect-size measure is appropriate for some
+#' situations. The user is completely responsible for specifying appropriate
+#' parameters (or for failing to do so)."
+#'
+#' In particular, effect size methods `"emmeans"` and `"bootES"` do not correct
+#' for covariates in the model, so should probably only be used when there is
+#' just one categorical predictor (with however many levels). If there are
+#' multiple predictors or any covariates, it is important to re-compute sigma
+#' adding back in the response variance associated with the variables that
```

Comment:
I know @bwiernik says this, but I am not convinced this -- as a blanket statement -- is true.
Adjusting predictions in multiple regression is something we do all the time - why is it valid for the means but not for the variance?

Reply:
Ok, I added "Some believe that", any suggestion as to how else I should change that paragraph? I suppose the intention is to try to warn users about the different risks involved...

``` r
#' In particular, effect size method `"bootES"` does not correct
#' for covariates in the model, so should probably only be used when there is
#' just one categorical predictor (with however many levels). Some believe that
#' if there are multiple predictors or any covariates, it is important to
#' re-compute sigma adding back in the response variance associated with the
#' variables that aren't part of the contrast.
```

Reply:
Thanks (:
It is perhaps more efficient to replace this whole section with "DON'T" 😜